### PR TITLE
Fix participant state machines after activation failures

### PR DESCRIPTION
**Pull request by languitar**
Fixes a few issues regarding the state machine implementation of participants in case activations failed. Should at least partially resolve https://code.cor-lab.org/issues/2775

### CODE REVIEW

File: rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java [hunk 1]
```diff
@@ -48,4 +55,35 @@ protected Participant createParticipant(
...
+     *
+     * @throws Exception
+     *             test error
+     */
+    @Test
+    public void deactivationAfterNewMethodFailure() throws Exception {
+        final LocalServer server = (LocalServer) createParticipant(
+                Utilities.createBrokenParticipantConfig());
+        server.activate();
+        try {
```
Comment (scymtym) [inline rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java]: Can we use this? https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html

File: rsb-java/src/test/java/rsb/patterns/RemoteServerStateTest.java [hunk 2]
```diff
@@ -48,4 +53,28 @@ protected Participant createParticipant(
...
+     *
+     * @throws Exception
+     *             test error
+     */
+    @Test
+    public void deactivationAfterNewMethodFailure() throws Exception {
+        final RemoteServer server = (RemoteServer) createParticipant(
+                Utilities.createBrokenParticipantConfig());
+        server.activate();
+        try {
```
Comment (scymtym) [inline rsb-java/src/test/java/rsb/patterns/RemoteServerStateTest.java]: See previous comment.

File: rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java (hunk 1 above)
Comment (languitar) [reply rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java]: This doesn't work here. `addMethod` and `deactivate` both can throw `RSBException`, but we only want to see it at the `addMethod` call. `deactivate` must still succeed. Using the exception annotation would make both indistinguishable and hence the test would also succeed in the unintended case that `deactivate` throws.

Comment (languitar) [reply rsb-java/src/test/java/rsb/patterns/LocalServerStateTest.java]: I will comment this

File: rsb-java/src/test/java/rsb/patterns/RemoteServerStateTest.java (hunk 2 above)
Comment (languitar) [reply rsb-java/src/test/java/rsb/patterns/RemoteServerStateTest.java]: as abovev

**Pull request merged**